lib/utils.js

```diff
@@ -142,12 +142,8 @@ api.fetchInSecurityContext = async ({
 }) => {
   if(url && typeof url === 'object' &&
     (url['@context'] === constants.SECURITY_CONTEXT_V2_URL ||
-      // skip compaction when using specific contexts, this is a temporary
-      // measure until a zcap context is created
-      isEqual(url['@context'], [
-        constants.SECURITY_CONTEXT_V2_URL,
-        'https://w3id.org/security/suites/ed25519-2020/v1'
-      ])
+     url['@context'] === ZCAP_CONTEXT_URL ||
+     Array.isArray(url['@context']) && url['@context'][0] === ZCAP_CONTEXT_URL
     )
   ) {
```

Comment:
IMO, I think the whole function needs to be cleaned up:

```js
  // fetch/assign zcap based on the given `url` which may be
  // a string or an object
  let zcap;
  if(typeof url === 'string') {
    ({document: zcap} = await documentLoader(url));
  } else if(url && typeof url === 'object' && url.id) {
    // root zcaps must always be dereferenced to avoid accepting
    // untrusted, user-provided input
    if(isRoot) {
      ({document: zcap} = await documentLoader(url.id));
    } else {
      // use user-provided object, `url`, as `zcap`
      zcap = url;
    }
    // ensure `url` is now a string
    url = url.id;
  }

  // ensure zcap is an object at this point
  if(zcap && typeof zcap === 'object') {
    throw new Error(`Could not fetch capability from url "${url}".`);
  }

  // determine if zcap must be compacted to security/v2 or if
  // it is already in a context that is understood by this library
  const ctx = zcap['@context'];
  if(ctx === constants.SECURITY_CONTEXT_V2_URL ||
    ctx === constants.ZCAP_CONTEXT_URL ||
    (Array.isArray(ctx) && ctx[0] === constants.ZCAP_CONTEXT_URL)) {
    return zcap;
  }

  return jsonld.compact(zcap, constants.SECURITY_CONTEXT_V2_URL, {
    documentLoader, expansionMap, compactToRelative: false
  });
```

Reply:
Updated here 31e2c3f. Changed `if(zcap && typeof zcap === 'object'` to
```js
if(zcap && typeof zcap !== 'object')
```

Reply:
Yeah, I got that wrong -- but the fix here is a still not quite right, I included the right fix in my comments. Thanks!